torchgeo/transforms/indices.py

```diff
@@ -22,119 +22,63 @@
-    """Compute Normalized Different Built-up Index (NDBI).
+class AppendNormalizedDifferenceIndex(Module):
+    """Append normalized difference index as channel to image tensor."""
 
-    Args:
-        swir: tensor containing swir band
-        nir: tensor containing nir band
-
-    Returns:
-        tensor containing computed NDBI values
-
-    """
-    return (swir - nir) / ((swir + nir) + _EPSILON)
-
-
-def nbr(nir: Tensor, swir: Tensor) -> Tensor:
-    """Compute Normalized Burn Ratio (NBR).
-
-    Args:
-        nir: tensor containing nir band
-        swir: tensor containing swir band
-
-    Returns:
-        tensor containing computed NBR values
-
-    .. versionadded:: 0.2.0
-    """
-    return (nir - swir) / ((nir + swir) + _EPSILON)
-
-
-def ndsi(green: Tensor, swir: Tensor) -> Tensor:
-    """Compute Normalized Different Snow Index (NDSI).
-
-    Args:
-        green: tensor containing green band
-        swir: tensor containing swir band
-
-    Returns:
-        tensor containing computed NDSI values
-    """
-    return (green - swir) / ((green + swir) + _EPSILON)
-
-
-def ndvi(red: Tensor, nir: Tensor) -> Tensor:
-    """Compute Normalized Different Vegetation Index (NDVI).
-
-    Args:
-        red: tensor containing red band
-        nir: tensor containing nir band
-
-    Returns:
-        tensor containing computed NDVI values
-    """
-    return (nir - red) / ((nir + red) + _EPSILON)
-
-
-def ndwi(green: Tensor, nir: Tensor) -> Tensor:
-    """Compute Normalized Different Water Index (NDWI).
-
-    Args:
-        green: tensor containing green band
-        nir: tensor containing nir band
-
-    Returns:
-        tensor containing computed NDWI values
-    """
-    return (green - nir) / ((green + nir) + _EPSILON)
-
-
-class AppendNDBI(Module):
-    """Normalized Difference Built-up Index (NDBI).
-
-    If you use this dataset in your research, please cite the following paper:
-
-    * https://doi.org/10.1080/01431160304987
-    """
-
-    def __init__(self, index_swir: int, index_nir: int) -> None:
+    def __init__(self, index_a: int, index_b: int) -> None:
         """Initialize a new transform instance.
 
         Args:
-            index_swir: index of the Short-wave Infrared (SWIR) band in the image
-            index_nir: index of the Near Infrared (NIR) band in the image
+            index_a: reference band channel index
+            index_b: difference band channel index
         """
         super().__init__()
         self.dim = -3
-        self.index_nir = index_nir
-        self.index_swir = index_swir
+        self.index_a = index_a
+        self.index_b = index_b
+
+    def _compute_index(self, band_a: Tensor, band_b: Tensor) -> Tensor:
+        """Compute normalized difference index.
+
+        Args:
+            band_a: reference band tensor
+            band_b: difference band tensor
+
+        Returns:
+            the index
+        """
+        return (band_a - band_b) / ((band_a + band_b) + _EPSILON)
```

Comment:
Is this helper function actually useful? It's only a one-liner and we only use it in one place, so it might make more sense to remove it.

Reply:
I think this is fine. It makes it really clear where the "compute index" logic is, and will be a good template for other classes to follow (that have more complicated logic).